<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>picProClass</class>
 <widget class="QMainWindow" name="picProClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1036</width>
    <height>567</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>qtest</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="6,0,1">
      <item>
       <widget class="QScrollArea" name="scrollArea">
        <property name="cursor" stdset="0">
         <cursorShape>CrossCursor</cursorShape>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents_2">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>816</width>
           <height>504</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="showImageLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QPushButton" name="OpenImageButton">
            <property name="text">
             <string>OpenImage</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="CloseImageButton">
            <property name="text">
             <string>CloseImage</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,1">
          <item>
           <widget class="QLabel" name="opLabel1">
            <property name="text">
             <string>Threshold:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="opText1">
            <property name="text">
             <string>1</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QSlider" name="opSlider1">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TicksAbove</enum>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_4">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,1">
          <item>
           <widget class="QLabel" name="opLabel2">
            <property name="text">
             <string>Threshold:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="opText2">
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_5">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QSlider" name="opSlider2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TicksAbove</enum>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_6">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <spacer name="verticalSpacer_7">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1036</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionCamera"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
   </widget>
   <widget class="QMenu" name="menuCornerPoint">
    <property name="title">
     <string>角点检测</string>
    </property>
    <addaction name="actionHarris"/>
    <addaction name="actionShiTomasi"/>
   </widget>
   <widget class="QMenu" name="menuFeature">
    <property name="title">
     <string>特征检测</string>
    </property>
    <widget class="QMenu" name="menuFeaturePoint">
     <property name="title">
      <string>特征点检测</string>
     </property>
     <addaction name="actionHARRIS"/>
     <addaction name="actionSURF"/>
     <addaction name="actionSIFT"/>
     <addaction name="actionFAST"/>
     <addaction name="actionORB"/>
     <addaction name="actionMSER"/>
     <addaction name="actionSTAR"/>
    </widget>
    <widget class="QMenu" name="menuFeatureMatch">
     <property name="title">
      <string>特征点匹配</string>
     </property>
     <addaction name="actionBruteForce"/>
     <addaction name="actionFlannBased"/>
     <addaction name="separator"/>
     <addaction name="actionBestMatch"/>
     <addaction name="actionKnnMatch"/>
    </widget>
    <widget class="QMenu" name="menuExtractor">
     <property name="title">
      <string>特征描述</string>
     </property>
     <addaction name="actionExtractorSIFT"/>
     <addaction name="actionExtractorSURF"/>
    </widget>
    <addaction name="menuFeaturePoint"/>
    <addaction name="menuExtractor"/>
    <addaction name="menuFeatureMatch"/>
    <addaction name="separator"/>
    <addaction name="actionObjectDetect"/>
    <addaction name="actionTempletMatch"/>
   </widget>
   <widget class="QMenu" name="menuBlur">
    <property name="title">
     <string>平滑处理</string>
    </property>
    <addaction name="actionBlur"/>
    <addaction name="actionGaussianBlur"/>
    <addaction name="actionMedianBlur"/>
    <addaction name="actionBilateralBlur"/>
   </widget>
   <widget class="QMenu" name="menuEdge">
    <property name="title">
     <string>边缘提取</string>
    </property>
    <addaction name="actionSobel"/>
    <addaction name="actionLaplace"/>
    <addaction name="actionCanny"/>
   </widget>
   <widget class="QMenu" name="menuThreshold">
    <property name="title">
     <string>阀值分割</string>
    </property>
    <widget class="QMenu" name="menuFixThres">
     <property name="title">
      <string>固定阀值</string>
     </property>
     <addaction name="actionBinary"/>
     <addaction name="actionBinaryInv"/>
     <addaction name="actionTrunc"/>
     <addaction name="actionTozero"/>
     <addaction name="actionTozeroInv"/>
    </widget>
    <addaction name="menuFixThres"/>
    <addaction name="actionAutoThreshold"/>
   </widget>
   <widget class="QMenu" name="menuPro">
    <property name="title">
     <string>简单处理</string>
    </property>
    <addaction name="actionResize"/>
    <addaction name="actionRotation"/>
    <addaction name="separator"/>
    <addaction name="actionDilation"/>
    <addaction name="actionErode"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuPro"/>
   <addaction name="menuBlur"/>
   <addaction name="menuThreshold"/>
   <addaction name="menuEdge"/>
   <addaction name="menuCornerPoint"/>
   <addaction name="menuFeature"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>打开</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>保存</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>新建</string>
   </property>
  </action>
  <action name="actionDilation">
   <property name="text">
    <string>膨胀</string>
   </property>
  </action>
  <action name="actionRecent">
   <property name="text">
    <string>最近</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>另存为</string>
   </property>
  </action>
  <action name="actionHarris">
   <property name="text">
    <string>Harris</string>
   </property>
  </action>
  <action name="actionShiTomasi">
   <property name="text">
    <string>Shi-Tomasi</string>
   </property>
  </action>
  <action name="actionErode">
   <property name="text">
    <string>腐蚀</string>
   </property>
  </action>
  <action name="actionResize">
   <property name="text">
    <string>缩放</string>
   </property>
  </action>
  <action name="actionRotation">
   <property name="text">
    <string>旋转</string>
   </property>
  </action>
  <action name="action_5">
   <property name="text">
    <string>旋转</string>
   </property>
  </action>
  <action name="action_9">
   <property name="text">
    <string>固定阀值</string>
   </property>
  </action>
  <action name="action_10">
   <property name="text">
    <string>自适应阀值</string>
   </property>
  </action>
  <action name="actionSobel">
   <property name="text">
    <string>Sobel</string>
   </property>
  </action>
  <action name="actionLaplace">
   <property name="text">
    <string>Laplace</string>
   </property>
  </action>
  <action name="actionCanny">
   <property name="text">
    <string>Canny</string>
   </property>
  </action>
  <action name="actionAutoThreshold">
   <property name="text">
    <string>自适应阀值</string>
   </property>
  </action>
  <action name="actionBlur">
   <property name="text">
    <string>均值平滑</string>
   </property>
  </action>
  <action name="actionGaussianBlur">
   <property name="text">
    <string>高斯平滑</string>
   </property>
  </action>
  <action name="actionMedianBlur">
   <property name="text">
    <string>中值平滑</string>
   </property>
  </action>
  <action name="actionBilateralBlur">
   <property name="text">
    <string>双边平滑</string>
   </property>
  </action>
  <action name="actionCamera">
   <property name="text">
    <string>拍照</string>
   </property>
  </action>
  <action name="actionObjectDetect">
   <property name="text">
    <string>平面物体检测</string>
   </property>
  </action>
  <action name="actionSURF">
   <property name="text">
    <string>SURF</string>
   </property>
  </action>
  <action name="actionSIFT">
   <property name="text">
    <string>SIFT</string>
   </property>
  </action>
  <action name="actionFAST">
   <property name="text">
    <string>FAST</string>
   </property>
  </action>
  <action name="actionFlannBased">
   <property name="text">
    <string>FlannBased</string>
   </property>
  </action>
  <action name="actionBruteForce">
   <property name="text">
    <string>BruteForce</string>
   </property>
  </action>
  <action name="actionORB">
   <property name="text">
    <string>ORB</string>
   </property>
  </action>
  <action name="actionHARRIS">
   <property name="text">
    <string>HARRIS</string>
   </property>
  </action>
  <action name="actionMSER">
   <property name="text">
    <string>MSER</string>
   </property>
  </action>
  <action name="actionSTAR">
   <property name="text">
    <string>STAR</string>
   </property>
  </action>
  <action name="actionExtractorSIFT">
   <property name="text">
    <string>SIFT</string>
   </property>
  </action>
  <action name="actionExtractorSURF">
   <property name="text">
    <string>SURF</string>
   </property>
  </action>
  <action name="actionTempletMatch">
   <property name="text">
    <string>模板匹配</string>
   </property>
  </action>
  <action name="actionFeaturePoint2">
   <property name="text">
    <string>点</string>
   </property>
  </action>
  <action name="actionFeatureMatch2">
   <property name="text">
    <string>匹配</string>
   </property>
  </action>
  <action name="actionObjectDetect2">
   <property name="text">
    <string>检测</string>
   </property>
  </action>
  <action name="actionBestMatch">
   <property name="text">
    <string>BestMatch</string>
   </property>
  </action>
  <action name="actionKnnMatch">
   <property name="text">
    <string>KnnMatch</string>
   </property>
  </action>
  <action name="actionBinary">
   <property name="text">
    <string>二进制</string>
   </property>
  </action>
  <action name="actionBinaryInv">
   <property name="text">
    <string>反二进制</string>
   </property>
  </action>
  <action name="actionTrunc">
   <property name="text">
    <string>截断</string>
   </property>
  </action>
  <action name="actionTozeroInv">
   <property name="text">
    <string>反阀值化为零</string>
   </property>
  </action>
  <action name="actionTozero">
   <property name="text">
    <string>阀值化为零</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="picpro.qrc"/>
 </resources>
 <connections/>
</ui>
